packages/odyssey-react-mui/src/Radio.tsx

```diff
@@ -95,8 +96,23 @@ const Radio = ({
   );
 
   const onChange = useCallback<NonNullable<MuiRadioProps["onChange"]>>(
-    (event, checked) => onChangeProp?.(event, checked),
-    [onChangeProp],
+    (event, checked) => {
+      if (isReadOnly) {
+        event.preventDefault();
```

Comment:
Can add `event.stopPropagation();` here if we want. May not be necessary but also won't hurt anything